include/net/net_ip.h

```diff
@@ -40,18 +40,24 @@ extern "C" {
 #define PF_UNSPEC	0	/* Unspecified.  */
 #define PF_INET		2	/* IP protocol family.  */
 #define PF_INET6	10	/* IP version 6.  */
+#define PF_PACKET	17	/* Packet family. */
 
 /** Address families.  */
 #define AF_UNSPEC	PF_UNSPEC
 #define AF_INET		PF_INET
 #define AF_INET6	PF_INET6
+#define AF_PACKET	PF_PACKET
 
 /** Protocol numbers from IANA */
 enum net_ip_protocol {
 	IPPROTO_ICMP = 1,
 	IPPROTO_TCP = 6,
 	IPPROTO_UDP = 17,
 	IPPROTO_ICMPV6 = 58,
+	IPPROTO_RAW = 255, /* As per IANA 255 is Reserved,
+			    * but RFC 3542 recommends 255.
```

Comment:
Ok, here we go, first normative reference! RFC 3542 is "Advanced Sockets Application Program Interface (API) for IPv6" and it indeed includes a chapter on raw sockets: https://tools.ietf.org/html/rfc3542#section-3

This information should go into PR description, to help reviewers, and in commit message(s).